tests/python/unittest/test_tvmscript_parser_source.py

```diff
@@ -0,0 +1,86 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unittests for tvm.script.parser.core"""
+import pytest
+import inspect
+import tvm.testing
+from tvm.script._parser.core.diagnostics import Source
+from tvm.script._parser.core import doc_core as doc
+from tvm.script import tir as T
+
+
+def matmul(a: T.handle, b: T.handle, c: T.handle) -> None:
+    A = T.match_buffer(a, [128, 128])
+    B = T.match_buffer(b, [128, 128])
+    C = T.match_buffer(c, [128, 128])
+    for i, j, k in T.grid(128, 128, 128):
+        with T.block("update"):
+            vi, vj, vk = T.axis.remap("SSR", [i, j, k])
+            C[vi, vj] = C[vi, vj] + A[vi, vk] * B[vj, vk]
+
+
+def test_source_base():
+    source = Source(matmul)
+    assert (
+        source.source_name == inspect.getsourcefile(matmul)
+        and source.start_line == 26
```

Comment:
Looks like we hardcoded the line number here, which might look a bit too ad hoc…perhaps we should just assert it’s not None in case of future changes to this file?

Reply:
Sure, fixed.